work/worker.go

```diff
@@ -289,11 +295,27 @@ func (self *worker) update() {
 	quitByErr := make(chan bool, 1)
 	go self.handleTxsCh(quitByErr)
 
+	// Initialize reset watchdog
+	if self.nodetype == common.CONSENSUSNODE && self.resetFn != nil && self.resetTimeOut > 0 {
+		callback := func() {
+			logger.Warn("Consensus timeout")
+			self.resetFn()
+		}
+		logger.Info("Initialize auto reset watchdog", "timeout", self.resetTimeOut.String())
+		self.resetWatchdog = time.AfterFunc(self.resetTimeOut, callback)
+		defer self.resetWatchdog.Stop()
+	}
+
 	for {
 		// A real event arrived, process interesting content
 		select {
 		// Handle ChainHeadEvent
 		case <-self.chainHeadCh:
+			// Refresh reset watchdog
```

Comment:
How about adding a new `case` with `time.Timer` instead of using `time.AfterFunc`?

Reply:
```
case <-resetTimer.C:
    self.restartFn() // It does nothing if an empty function is set
```

Reply:
I thought this routine is simple.
Also if I add the watchdog reset routine in case statements, the watchdog can't work properly when the loop has a problem.